torch/fx/experimental/symbolic_shapes.py

```diff
@@ -5389,22 +5389,35 @@ def _maybe_evaluate_static(
 
         # axioms with compute hint NYE
         assert not compute_hint or not axioms
-
         expr = self.simplify(expr)
 
         if compute_hint:
             expr = expr.xreplace(self.var_to_val).xreplace(self.unbacked_var_to_val)
 
         expr = canonicalize_bool_expr(expr)
 
+        def try_simplify_floor_div(axioms):
+            for k, v in axioms.items():
+                # A FloorDiv in implications could have became CleanDiv at this point, due to new facts
+                # to the sahpeEnv. This handles such issue but its not ideal. This is the only expression
+                # simplification that depends on the global state of shape env.
+                # TODO try to get rid of CleanDiv since its breaks the invariant thats simplifications of sympy
+                # expressions only depend on the expression itself.
+                if k.has(FloorDiv) and k.free_symbols.issubset(expr.free_symbols):
```

Comment:
check if its more efificient to remove 
and k.free_symbols.issubset(expr.free_symbols):